src/graph_ops.rs

```diff
@@ -24,7 +24,7 @@ pub trait InDegreePartitionOp<Node: Idx> {
 pub trait ForEachNodeOp<Node: Idx> {
     fn for_each_node<T, F>(
         &self,
-        partition: &[Range<Node>],
+        maybe_partition: Option<&[Range<Node>]>,
```

Comment:
I think it's not so common to use `maybe_` as prefix for `Option`. We use it in Java because we can't shadow variable names, which we can do here :)